runtime/path.go

```diff
@@ -0,0 +1,329 @@
+package runtime
```

Comment:
Is this just a copy of the `query.go` file?

Reply:
Yes, pretty much, I just modified it to accept `map[string]string` instead of `http.Request`. Is that fine? lol

Reply:
Sure, but we don't necessarily want to parse all the same values as we do in query parameters - for example maps and repeated types cannot be set in path parameters.

Reply:
I think I disabled those already, cool, will make sure, thanks!

Reply:
It seems repeated enums can be set yes? I'll add those into the parser.

Reply:
Care to provide additional information? I don't know what you're referring to. The original grpc-gateway template code supports repeated enums as a path parameter from what I could tell.

Reply:
I mean, I just don't understand how one could add repeated enums into a path. Do you know? What would it look like?

Reply:
I could be missing context but I see support for it here: https://github.com/grpc-ecosystem/grpc-gateway/blob/master/protoc-gen-grpc-gateway/internal/gengateway/template.go#L367

Maybe this is for general params instead of pathParams. Its not helpful that the source of all values here is coming from a map named `pathParams` so maybe I misunderstood how this is used.